brainbeats-backend/Controllers/BeatController.cs

```diff
@@ -13,189 +14,124 @@ public class BeatController : ControllerBase
     [HttpPost]
     [Route("create_beat")]
     public async Task<IActionResult> CreateBeat(dynamic req) {
-      string request = GetRequest(req);
-      var body = JsonConvert.DeserializeObject<dynamic>(request);
-
-      string beatId = Guid.NewGuid().ToString();
-      string name = body.name;
-      string email = body.email;
-
-      string duration = body.duration;
-      string image = body.image;
-      string isPrivate = body.isPrivate;
-      string instrumentList = body.instrumentList;
-      string createdDate = GetCurrentTime();
-      string modifiedDate = GetCurrentTime();
-      string attributes = body.attributes;
-      string audio = body.audio;
-      string seed = body.seed;
-
-      StringBuilder queryString = new StringBuilder();
-
-      try {
-        queryString.Append(CreateVertex("beat", beatId) +
-          AddProperty("name", name) +
-          AddProperty("duration", duration) +
-          AddProperty("image", image) +
-          AddProperty("isPrivate", isPrivate) +
-          AddProperty("instrumentList", instrumentList) +
-          AddProperty("createdDate", createdDate) +
-          AddProperty("modifiedDate", modifiedDate) +
-          AddProperty("attributes", attributes) +
-          AddProperty("audio", audio) +
-          CreateEdge("OWNED_BY", email));
-      } catch {
-        return BadRequest("Malformed Request");
-      }
-
-      if (seed != null) {
-        queryString.Append(EdgeSourceReference() + AddProperty("seed", seed, false));
+      JObject body = DeserializeRequest(req);
+
+      string queryString;
+
+      try {
+        string beatId = Guid.NewGuid().ToString();
+        List<KeyValuePair<string, string>> edges = new List<KeyValuePair<string, string>> {
+          new KeyValuePair<string, string>("OWNED_BY", body.GetValue("email").ToString())
+        };
+
+        queryString = CreateVertexQuery("beat", beatId, body, edges);
+      } catch {
+        return BadRequest("Malformed Request");
       }
 
       try {
         var result = await DatabaseConnection.Instance.ExecuteQuery(queryString.ToString());
         return Ok(result);
       } catch {
-        return BadRequest();
+        return BadRequest("Something went wrong");
```

Comment:
Is it possible to give a more helpful error message?

Reply:
Will create a new issue to enable Azure Application Insights -> This is too large of a change to include in this PR